test/tests_for_integration/api_to_db_test.py

```diff
@@ -763,3 +763,161 @@ def _run_job_concierge(
+    # this error could probably use some cleanup
+    err = (
+        "('An error occurred while fetching user permissions from the Workspace', "
+        + "ServerError('No workspace with id 1 exists'))"
+    )
+    _run_job_concierge_fail(ee2_port, TOKEN_KBASE_CONCIERGE, params, {"a": "b"}, err)
+
+
+def test_run_job_concierge_fail_not_concierge(ee2_port):
+    params = {"method": "mod.meth", "app_id": "mod/app"}
+    err = "You are not the concierge user. This method is not for you"
+    _run_job_concierge_fail(ee2_port, TOKEN_NO_ADMIN, params, {"a": "b"}, err)
+
+
+def test_run_job_concierge_fail_bad_method(ee2_port):
+    params = {"method": "mod.meth.moke", "app_id": "mod/app"}
+    err = "Unrecognized method: 'mod.meth.moke'. Please input module_name.function_name"
+    _run_job_concierge_fail(ee2_port, TOKEN_KBASE_CONCIERGE, params, {"a": "b"}, err)
+
+
+def test_run_job_concierge_fail_reqs_list_not_list(ee2_port):
+    params = {"method": "mod.meth", "app_id": "mod/app"}
+    conc_params = {"requirements_list": {"a": "b"}}
+    err = "requirements_list must be a list"
+    _run_job_concierge_fail(ee2_port, TOKEN_KBASE_CONCIERGE, params, conc_params, err)
+
+
+def test_run_job_concierge_fail_reqs_list_bad_req(ee2_port):
+    params = {"method": "mod.meth", "app_id": "mod/app"}
+    conc_params = {"requirements_list": ["a=b", "touchmymonkey"]}
+    err = "Found illegal requirement in requirements_list: touchmymonkey"
+    _run_job_concierge_fail(ee2_port, TOKEN_KBASE_CONCIERGE, params, conc_params, err)
+
+
+def test_run_job_concierge_fail_bad_cpu(ee2_port):
+    params = {"method": "mod.meth", "app_id": "mod/app"}
+    conc_params = {"request_cpus": [2]}
+    err = (
+        "Found illegal cpu request '[2]' in job requirements from concierge parameters"
+    )
+    _run_job_concierge_fail(ee2_port, TOKEN_KBASE_CONCIERGE, params, conc_params, err)
+
+
+def test_run_job_concierge_fail_bad_mem(ee2_port):
+    params = {"method": "mod.meth", "app_id": "mod/app"}
+    conc_params = {"request_memory": "-3"}
+    err = "memory in MB must be at least 1"
+    _run_job_concierge_fail(ee2_port, TOKEN_KBASE_CONCIERGE, params, conc_params, err)
+
+
+def test_run_job_concierge_fail_bad_disk(ee2_port):
+    params = {"method": "mod.meth", "app_id": "mod/app"}
+    conc_params = {"request_disk": 4.5}
+    err = (
+        "Found illegal disk request '4.5' in job requirements from concierge parameters"
+    )
+    _run_job_concierge_fail(ee2_port, TOKEN_KBASE_CONCIERGE, params, conc_params, err)
+
+
+def test_run_job_concierge_fail_bad_clientgroup(ee2_port):
+    params = {"method": "mod.meth", "app_id": "mod/app"}
+    conc_params = {"client_group": "fakefakefake"}
+    err = "No such clientgroup: fakefakefake"
+    _run_job_concierge_fail(ee2_port, TOKEN_KBASE_CONCIERGE, params, conc_params, err)
+
+
+def test_run_job_concierge_fail_bad_clientgroup_regex(ee2_port):
+    params = {"method": "mod.meth", "app_id": "mod/app"}
```

Comment:
Can we DRY with the params that are the same?

Reply:
done

Reply:
I totally did not expect that, I expected the entire "params" to be saved and re-used, but this is good too.